server/backends/disk_cache/disk_cache.go

```diff
@@ -101,35 +101,48 @@ func (c *DiskCache) WithPrefix(prefix string) interfaces.Cache {
 	}
 
 	return &DiskCache{
-		l:       c.l,
-		rootDir: c.rootDir,
-		lock:    c.lock,
-		prefix:  newPrefix,
+		l:            c.l,
+		rootDir:      c.rootDir,
+		lock:         c.lock,
+		prefix:       newPrefix,
+		diskIsMapped: c.diskIsMapped,
 	}
 }
 
 func (c *DiskCache) initializeCache() error {
 	if err := disk.EnsureDirectoryExists(c.rootDir); err != nil {
 		return err
 	}
-	records := make([]*fileRecord, 0)
-	walkFn := func(path string, info os.FileInfo, err error) error {
-		if !info.IsDir() {
-			records = append(records, makeRecord(path, info))
+	c.diskIsMapped = false
+	go func() {
+		start := time.Now()
+		records := make([]*fileRecord, 0)
+		walkFn := func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() {
+				records = append(records, makeRecord(path, info))
+			}
+			return nil
+		}
+		if err := filepath.Walk(c.rootDir, walkFn); err != nil {
```

Comment:
go 1.16 has filepath.WalkDir which is supposed to be more efficient

Reply:
I tested this in another CL and didn't notice much difference but I've switched to it.